File: hackerrank/insertion-sort/insertion-sort-pt-1/solution.py
def insertionSort1(n, arr):
    last_val = arr[n - 1]
    should_return = False

    for i in reversed(range(0, n - 1)):
        if arr[i] > last_val:
            arr[i + 1] = arr[i]
        else:
            arr[i + 1] = last_val
            should_return = True

        print(*arr, sep=" ")

        if (should_return):
            break

    if not should_return:
        arr[0] = last_val
        print(*arr, sep=" ")

    return

File: hackerrank/insertion-sort/insertion-sort-pt-1/solution.cpp
void insertionSort1(int n, vector<int> arr) {
    bool should_return = false;
    int last_value = arr[n - 1];

    for (int i = n - 2; i >= 0; i--) {
        if (arr[i] > last_value) {
            arr[i + 1] = arr[i];
        } else {
            arr[i + 1] = last_value;
            should_return = true;
        }

        for (int i = 0; i < n; i++) {
            printf("%d ", arr[i]);
        }

        printf("\n");

        if (should_return) break;
    }

    if (!should_return) {
        arr[0] = last_value;

        for (int i = 0; i < n; i++) {
            printf("%d ", arr[i]);
        }

        printf("\n");
    }

    return;
}
